### DISCUSSION TITLE
QR modal not showing on app deployed to IPFS

### DISCUSSION THREAD

**Discussion by torreyatcitty** [WalletConnect/walletconnect-monorepo#2814] [Developer Support]
Currently made the changes to upgrade wallet connect to v2 in our Dapp.

What we've noticed though, is everything appears to work fine locally. We can scan QR codes and run transactions. However, when we deploy our build to IPFS, we are seeing an error when attempting to call `connector.activate()` with `showQrModal` set to true when we initialize WalletConnectV2.

We see the following error on a build deployed to IPFS

```
main.3917c7d5.js:166 Uncaught (in promise) TypeError: w.default.init is not a function
    at TA.createClient (main.3917c7d5.js:166:30561)
    at TA.initialize (main.3917c7d5.js:166:30415)
    at TA.init (main.3917c7d5.js:166:27620)
    at H.initialize (main.3917c7d5.js:166:53459)
    at H.init (main.3917c7d5.js:166:49396)
    at t.<anonymous> (main.3917c7d5.js:166:57960)
    at Generator.next (<anonymous>)
    at d (main.3917c7d5.js:84:1871)
```
   
Happy to provide more details, but not sure what is happening.

**Comment by crypblizz8** [comment]
Hey @torreyatcitty, great to hear you are migrating. 

Can you provide more details on:
1) Version of the package you are using
2) Reproducible link 
3) Which wallet (platform and version) you used to get to this error

**Comment by torreyatcitty** [comment]
Thanks for getting back to me @crypblizz8 .

1. Here are the relevant package versions we are using:
```
    "@web3-react/core": "8.2.0",
    "@web3-react/eip1193": "8.2.0",
    "@web3-react/empty": "8.2.0",
    "@web3-react/metamask": "8.2.0",
    "@web3-react/network": "8.2.0",
    "@web3-react/types": "8.2.0",
    "@web3-react/url": "8.2.0",
    "@web3-react/walletconnect": "^8.2.0",
    "@web3-react/walletconnect-v2": "^8.3.6",
```

2. Here is a link with a build that shows the problem:
https://app-v3-ipfs.infura-ipfs.io/ipfs/QmSdERezrqaD5mLdLL6ARviAz1KZy6R4iP3qpZiwxuo4ox/

3. The QR code doesn't show up when we try to select the WalletConnect option. We are given the error above in console logs. (Note the same code works locally just fine, QRCode shows up)

Lastly, we are using the web3-react version of initializing the WallectConnectV2 connect like this:
```
const walletConnectConnector = initializeConnector<WalletConnectV2>(
  (actions) =>
    new WalletConnectV2({
      actions,
      options: {
        projectId: WALLECT_CONNECT_PROJECT_ID,
        chains: requiredChains,
        optionalChains,
        showQrModal: true,
      },
    })
);
```

**Comment by chris13524** [comment]
A workaround and potential fix has been identified in the [web3-react repo](https://github.com/Uniswap/web3-react/issues/840#top). Closing this discussion to focus conversation there!
